### ISSUE TITLE
HypermediaSupportBeanDefinitionRegistrar double registers message converters

### ISSUE THREAD

**Issue by gregturn** [spring-projects/spring-hateoas#558]
The follow code inside `HypermediaSupportBeanDefinitionRegistrar` double registers the message converters:

```
@Override
public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {

	if (bean instanceof RequestMappingHandlerAdapter) {

		RequestMappingHandlerAdapter adapter = (RequestMappingHandlerAdapter) bean;
		adapter.setMessageConverters(potentiallyRegisterModule(adapter.getMessageConverters()));
	}

	if (bean instanceof RequestMappingHandlerAdapter) {

		RequestMappingHandlerAdapter adapter = (RequestMappingHandlerAdapter) bean;
		adapter.setMessageConverters(potentiallyRegisterModule(adapter.getMessageConverters()));
	}

	if (bean instanceof RestTemplate) {

		RestTemplate template = (RestTemplate) bean;
		template.setMessageConverters(potentiallyRegisterModule(template.getMessageConverters()));
	}

	return bean;
}
```

**Comment by olivergierke** [comment]
It doesn't really double register them as the second `potentiallyRegisterModule(…)` should actually detect the one already registered in the first call. But yeah, it's definitely obsolete code and could be removed. Probably a leftover from the days we also dealt with an `AnnotationMethodHandlerAdapter`.

**Comment by olivergierke** [comment]
Fixed with #568.
